src/renderer/examples/main.rs

```diff
@@ -0,0 +1,234 @@
+use amethyst_renderer::{ColorFormat, DepthFormat};
+
+fn build_sphere() -> Vec<Vertex> {
+    SphereUV::new(32, 32)
+        .vertex(|(x, y, z)| Vertex{
+            pos: [x, y, z],
+            normal: Vector3::new(x, y, z).normalize().into()
+        })
+        .triangulate()
+        .vertices()
+        .collect()
+}
+
+fn main() {
+    let builder = glutin::WindowBuilder::new()
+        .with_title("Amethyst Renderer Demo".to_string())
+        .with_dimensions(800, 600)
+        .with_vsync();
+
+    let (window, mut device, mut factory, main_color, main_depth) =
+        gfx_window_glutin::init::<ColorFormat, DepthFormat>(builder);
+    let combuf = factory.create_command_buffer();
+
+    let sphere = build_sphere();
+    let buffer = factory.create_vertex_buffer(&sphere);
+    let slice = gfx::Slice::new_match_vertex_buffer(&buffer);
+
+    let mut scene = amethyst_renderer::Scene{
+        fragments: vec![],
+        lights: vec![]
+    };
+
+    let mut rng = rand::thread_rng();
+
+    for x in -1..2 {
+        for y in -1..2 {
+            for z in -1..2 {
+                let x = x as f32 * 4.;
+                let y = y as f32 * 4.;
+                let z = z as f32 * 4.;
+
+                let color = [rng.gen_range(0., 1.), rng.gen_range(0., 1.), rng.gen_range(0., 1.), 1.];
+
+                scene.fragments.push(amethyst_renderer::Fragment{
+                    buffer: buffer.clone(),
+                    slice: slice.clone(),
+                    ka: [color[0] * 0.1, color[1] * 0.1, color[2] * 0.1, 1.],
+                    kd: color,
+                    transform: Matrix4::from_translation(Vector3::new(x, y, z)).into()
+                })
+            }
+        }
+    }
+
+    for x in -2..3 {
+        for y in -2..3 {
+            for z in -2..3 {
+                let x = x as f32 * 5.;
+                let y = y as f32 * 5.;
+                let z = z as f32 * 5.;
+
+                let r = (x + 10.) / 20.;
+                let g = (y + 10.) / 20.;
+                let b = (z + 10.) / 20.;
+
+                scene.lights.push(amethyst_renderer::Light{
+                    color: [r, g, b, 1.],
+                    radius: 1.,
+                    center: [x, y, z],
+                    propagation_constant: 0.,
+                    propagation_linear: 0.,
+                    propagation_r_square: 1.,
+                })
+            }
+        }
+    }
+
+
+    let mut frame = amethyst_renderer::Frame{
+        passes: vec![],
+        targets: HashMap::new(),
+        scenes: std::collections::HashMap::new(),
+        cameras: std::collections::HashMap::new()
+    };
+
+    frame.scenes.insert(format!("main"), scene);
+    frame.passes.push(amethyst_renderer::RenderPasses::new(format!("gbuffer")));
+    frame.passes[0].operations = vec![
+        Box::new(amethyst_renderer::Clear{color: [0., 0., 0., 1.]}),
+        Box::new(amethyst_renderer::Draw{
+            camera: format!("main"),
```

Comment:
There is too much `format!` noise. I understand that we need to receive `String` there, but maybe we can help it by adding some constructors like `amethyst_render::Draw::new("main", "main")`?